src/backend/distributed/commands/foreign_constraint.c

```diff
@@ -882,17 +819,25 @@ GetForeignKeyOids(Oid relationId, int flags)
 	Assert(!(extractReferencing && extractReferenced));
 	Assert(extractReferencing || extractReferenced);
 
+	bool extractDistributed = (flags & ONLY_DISTRIBUTED_TABLES);
+	bool extractReference = (flags & ONLY_REFERENCE_TABLES);
+	bool extractCitusLocal = (flags & ONLY_CITUS_LOCAL_TABLES);
+
+	Assert(!(extractDistributed && extractReference));
+	Assert(!(extractDistributed && extractCitusLocal));
+	Assert(!(extractReference && extractCitusLocal));
```

Comment:
I don't see a reason for this

Reply:
After removing those three, maybe we should rename those new flags to `FILTER_XXX_TABLE`

Reply:
The new flags filter other tables out. So if you have both `ONLY_DISTRIBUTED_TABLES` and `ONLY_REFERENCE_TABLES` every table will be filtered out. I could change it by changing my code below but the reason I did it this way is to keep original behavior of the `GetForeignKeyOids` same.  I would need to add all the flags everywhere this functions was called. But there was no code that needed more than one of the table types, so I didn't bother.

Reply:
then makes sense keep it as is, but might worth to comment those three assertions then

Reply:
Why not make these new flags inclusive as well and adjust the callers?

Reply:
Mainly because I didn't think it was necessary. There weren't a need for filtering more than one table types (currently). So I didn't think I should change the default behavior and update all the places the function is called.

But if you're thinking it would be a better solution, it is definitely doable.

Reply:
I mainly just find the combination of inclusive and exclusive confusing, e.g.:
```c
INCLUDE_REFERENCING_CONSTRAINTS | ONLY_REFERENCE_TABLES
```